### PR TITLE
Import rocprofiler

### PR DESCRIPTION
**Pull request by littlewu2508**
Introduce dev-util/rocprofiler as the ROCm profiler. Support version 4.3.0 and 5.0.2.

Patches applied to dev-util/roctracer to compile libtracer_tool.so.

Patches applied to dev-util/rocprofiler and dev-util/roctracer to avoid load libhsa-amd-aqlprofile64.so.

### CODE REVIEW

**Comment by heroxbd** [comment]
Please unify your names between the commit author and signing-offs.

**Comment by littlewu2508** [comment]
Done
--

**Comment by heroxbd** [comment]
[build.log](https://github.com/gentoo/gentoo/files/8885725/build.log)
I am not able to compile roctracer.

**Comment by heroxbd** [comment]
From the `build.log`
```
cc1: fatal error: /usr/lib/hip/include/hip/hip_runtime_api.h: No such file or directory
```

but with HIP 5.0.2
```
$ equery b hip_runtime_api.h
 * Searching for hip_runtime_api.h ... 
dev-util/hip-5.0.2 (/usr/include/hip/hip_runtime_api.h)
```

**Comment by littlewu2508** [comment]
> From the `build.log`
> 
> ```
> cc1: fatal error: /usr/lib/hip/include/hip/hip_runtime_api.h: No such file or directory
> ```
> 
> but with HIP 5.0.2
> 
> ```
> $ equery b hip_runtime_api.h
>  * Searching for hip_runtime_api.h ... 
> dev-util/hip-5.0.2 (/usr/include/hip/hip_runtime_api.h)
> ```

My bad. I should first fix dev-util/hip's hipconfig and use that to determine HIP_PATH

**Comment by littlewu2508** [comment]
@heroxbd Please check whether the new commit fixes the issue. I can compile the roctracer normally.

File: dev-util/roctracer/roctracer-5.0.2.ebuild [hunk 1]
```diff
@@ -44,22 +46,29 @@ src_prepare() {
...
 		-e "/add_subdirectory ( \${HSA_TEST_DIR} \${PROJECT_BINARY_DIR}/d" \
+		-e "/DESTINATION/s,\${DEST_NAME}/tool,$(get_libdir),g" \
 		-i test/CMakeLists.txt || die
 
 	hprefixify script/*.py
 }
 
 src_configure() {
-	export HIP_PATH="${EPREFIX}/usr"
+	export HIP_PATH="${EPREFIX}/usr/lib/hip"
```
Comment (heroxbd) [inline dev-util/roctracer/roctracer-5.0.2.ebuild]: What's this?

**Review by heroxbd — CHANGES_REQUESTED** [review]
It is not obvious how you patched out the aqlprofile loading.  The changes in the commit look completely irrelevant.

File: dev-util/rocprofiler/files/rocprofiler-4.3.0-gentoo-location.patch [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+ DATA_DIR="rpl_data_${time_stamp}_$$"
+ 
+-RPL_PATH=$PKG_DIR/lib
+-TLIB_PATH=$PKG_DIR/tool
+-TTLIB_PATH=$TT_DIR/tool
+-ROCM_LIB_PATH=$ROOT_DIR/lib
++RPL_PATH=$PKG_DIR/lib64
++TLIB_PATH=$PKG_DIR/lib64
++TTLIB_PATH=$TT_DIR/lib64
++ROCM_LIB_PATH=$ROOT_DIR/lib64
```
Comment (heroxbd) [inline dev-util/rocprofiler/files/rocprofiler-4.3.0-gentoo-location.patch:24]: Should not hard-code `lib64`.

File: dev-util/rocprofiler/files/rocprofiler-5.0.2-gentoo-location.patch [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+ DATA_DIR="rpl_data_${time_stamp}_$$"
+ 
+-RPL_PATH=$PKG_DIR/lib
+-TLIB_PATH=$PKG_DIR/tool
+-TTLIB_PATH=$TT_DIR/tool
+-ROCM_LIB_PATH=$ROOT_DIR/lib
++RPL_PATH=$PKG_DIR/lib64
++TLIB_PATH=$PKG_DIR/lib64
++TTLIB_PATH=$TT_DIR/lib64
++ROCM_LIB_PATH=$ROOT_DIR/lib64
```
Comment (heroxbd) [inline dev-util/rocprofiler/files/rocprofiler-5.0.2-gentoo-location.patch:24]: same.

**Review by heroxbd — CHANGES_REQUESTED** [review]

File: dev-util/roctracer/roctracer-5.0.2.ebuild (hunk 1 above)
Comment (heroxbd) [reply dev-util/roctracer/roctracer-5.0.2.ebuild]: I should've have reviewed this diff.  Please squash and reorganize your commits into logical units.

**Review by heroxbd — CHANGES_REQUESTED** [review]

File: dev-util/rocprofiler/files/rocprofiler-4.3.0-gentoo-location.patch (hunk 2 above)
Comment (littlewu2508) [reply dev-util/rocprofiler/files/rocprofiler-4.3.0-gentoo-location.patch:24]: Is there a decent solution like `@GENTOO_PORTAGE_EPREFIX@`?

File: dev-util/roctracer/roctracer-5.0.2.ebuild (hunk 1 above)
Comment (littlewu2508) [reply dev-util/roctracer/roctracer-5.0.2.ebuild]: My mistake. I'll rebase it out.

File: dev-util/rocprofiler/files/rocprofiler-4.3.0-gentoo-location.patch (hunk 2 above)
Comment (littlewu2508) [reply dev-util/rocprofiler/files/rocprofiler-4.3.0-gentoo-location.patch:24]: I'll try patch it into @LIB_DIR@ and use sed

**Comment by heroxbd** [comment]
littlewu2508 ***@***.***> writes:

> I'll try patch it into @LIB_DIR@ and use sed

Yes, that's the way to go.  It also alignes with `autoconf`.

**Pull request closed without merging**